tests/integration/pdo/mysql/base-class/test_instance_reporting_1.php

```diff
@@ -0,0 +1,157 @@
+Supportability/TxnData/SlowSQL, 1
+*/
+
+/*EXPECT_SLOW_SQLS
+[
+  [
+    [
+      "OtherTransaction/php__FILE__",
+      "<unknown>",
+      "?? SQL id",
+      "DROP TABLE ENV[DATASTORE_COLLECTION];",
+      "Datastore/statement/ENV[DATASTORE_PRODUCT]/ENV[DATASTORE_COLLECTION]/drop",
+      1,
+      "?? total time",
+      "?? min time",
+      "?? max time",
+      {
+        "backtrace": [
+          "/ in PDO::exec called at .*\/",
+          " in test_instance_reporting called at __FILE__ (??)"
+        ],
+        "host": "ENV[MYSQL_HOST]",
+        "port_path_or_id": "ENV[MYSQL_PORT]",
+        "database_name": "ENV[MYSQL_DB]"
+      }
+    ],
+    [
+      "OtherTransaction/php__FILE__",
+      "<unknown>",
+      "?? SQL id",
+      "CREATE TABLE ENV[DATASTORE_COLLECTION] (id INT, description VARCHAR(?));",
+      "Datastore/statement/ENV[DATASTORE_PRODUCT]/ENV[DATASTORE_COLLECTION]/create",
+      1,
+      "?? total time",
+      "?? min time",
+      "?? max time",
+      {
+        "backtrace": [
+          "/ in PDO::exec called at .*\/",
+          " in test_instance_reporting called at __FILE__ (??)"
+        ],
+        "host": "ENV[MYSQL_HOST]",
+        "port_path_or_id": "ENV[MYSQL_PORT]",
+        "database_name": "ENV[MYSQL_DB]"
+      }
+    ]
+  ]
+]
+*/
+
+/*EXPECT_SPAN_EVENTS_LIKE
+[
+  [
+    {
+      "category": "datastore",
+      "type": "Span",
+      "guid": "??",
+      "traceId": "??",
+      "transactionId": "??",
+      "name": "Datastore/statement/ENV[DATASTORE_PRODUCT]/ENV[DATASTORE_COLLECTION]/create",
+      "timestamp": "??",
+      "duration": "??",
+      "priority": "??",
+      "sampled": true,
+      "parentId": "??",
+      "span.kind": "client",
+      "component": "ENV[DATASTORE_PRODUCT]"
+    },
+    {},
+    {
+      "peer.hostname": "ENV[MYSQL_HOST]",
+      "peer.address": "ENV[MYSQL_HOST]:ENV[MYSQL_PORT]",
+      "db.instance": "ENV[MYSQL_DB]",
+      "db.statement": "CREATE TABLE ENV[DATASTORE_COLLECTION] (id INT, description VARCHAR(?));"
+    }
+  ],
+  [
+    {
+      "category": "datastore",
+      "type": "Span",
+      "guid": "??",
+      "traceId": "??",
+      "transactionId": "??",
+      "name": "Datastore/statement/ENV[DATASTORE_PRODUCT]/ENV[DATASTORE_COLLECTION]/drop",
+      "timestamp": "??",
+      "duration": "??",
+      "priority": "??",
+      "sampled": true,
+      "parentId": "??",
+      "span.kind": "client",
+      "component": "ENV[DATASTORE_PRODUCT]"
+    },
+    {},
+    {
+      "peer.hostname": "ENV[MYSQL_HOST]",
+      "peer.address": "ENV[MYSQL_HOST]:ENV[MYSQL_PORT]",
+      "db.instance": "ENV[MYSQL_DB]",
+      "db.statement": "DROP TABLE ENV[DATASTORE_COLLECTION];"
+    }
+  ]
+]
+*/
+
+require_once(realpath (dirname ( __FILE__ )) . '/../../test_instance_reporting.inc');
+require_once(realpath (dirname ( __FILE__ )) . '/../../../../include/config.php');
+
+test_instance_reporting(new PDO($PDO_MYSQL_DSN, $MYSQL_USER, $MYSQL_PASSWD), 0);
```

Comment:
The descriptions between this test and the test_instance_reporting_2.php are the same, but the code to connect is different here than for [reporting_2](https://github.com/newrelic/newrelic-php-agent/pull/993/files#diff-a28747bcacf6e28f41e0a8e6b95933d73810cfabc008fde280cf55ed9d815b31R156-R160).

Could the description be updated to explain the difference? And why the different way is needed?

Same comment to all reporting_1 vs reporting_2 test cases.
I see you've broken out the directories by base-class, constructor, factory, etc, but the reporting_2 cases for pdo/mysql all still reference base class and have different connection blocks than the rest of the pdo/mysql tests in their respective directories.
https://github.com/newrelic/newrelic-php-agent/pull/993/files#diff-ee3c25291b81a5eb33e3e6e95f4d3bb0c1e9c817001a1bc88af0ab8acc31453cR8
https://github.com/newrelic/newrelic-php-agent/pull/993/files#diff-9465b3b1aacc6abb291baf0a70e282de761c1fa60199e26771cc11e9fa6338a9R8

Reply:
The code to connect is the same and depends on the location of the test: base-class uses PDO base class constructor (`new PDO(...)`), factory uses static factory method from `PDO` class (`PDO::Connect(...)`), and constructor uses driver specific subclass of PDO base class (`new Pdo\{dbdriver}(...)`). What is different between test_instance_reporting_1 and test_instance_reporting_2 is DSN used in each of those methods: test_instance_reporting_1 uses remote host and TCP port vs test_instance_reporting_2 uses unix socket on local host and this difference is already reflected in tests' descriptions here: [1](https://github.com/newrelic/newrelic-php-agent/blob/feat/php-8.4/fix-pdo-instrumentation/tests/integration/pdo/mysql/base-class/test_instance_reporting_1.php#L9) vs [2](https://github.com/newrelic/newrelic-php-agent/blob/feat/php-8.4/fix-pdo-instrumentation/tests/integration/pdo/mysql/base-class/test_instance_reporting_2.php#L9).

test_instance_reporting_1 and test_instance_reporting_2 differ in test expectation: see value of `"peer.hostname"` span's attribute in [1](https://github.com/newrelic/newrelic-php-agent/blob/feat/php-8.4/fix-pdo-instrumentation/tests/integration/pdo/mysql/base-class/test_instance_reporting_1.php#L146) vs [2](https://github.com/newrelic/newrelic-php-agent/blob/feat/php-8.4/fix-pdo-instrumentation/tests/integration/pdo/mysql/base-class/test_instance_reporting_2.php#L145) as well as value of `"port_path_or_id"` in slow SQL trace [1](https://github.com/newrelic/newrelic-php-agent/blob/feat/php-8.4/fix-pdo-instrumentation/tests/integration/pdo/mysql/base-class/test_instance_reporting_1.php#L93) vs [2](https://github.com/newrelic/newrelic-php-agent/blob/feat/php-8.4/fix-pdo-instrumentation/tests/integration/pdo/mysql/base-class/test_instance_reporting_2.php#L92).